frontend/packages/knative-plugin/src/components/add/eventSource-validation-utils.ts

```diff
@@ -69,8 +69,19 @@ export const sourceDataSpecSchema = yup
         topics: yup.string().required('Required'),
       }),
     }),
+  })
+  .when('type', {
+    is: EventSources.ContainerSource,
+    then: yup.object().shape({
+      containersource: yup.object().shape({
+        containers: yup.array().of(
+          yup.object({
```

Comment:
I think here also it should be `yup.object().shape({image: yup.string().required('Required')})`

Reply:
You can either use `yup.objec().shape({x: y})` or directly pass the shape in the object's constructor `yup.object({x:y})`. Both does exactly the same thing.
Refer: https://github.com/jquense/yup#object
Usually, i do object().shape if there is nested object, for plain one level object i pass the shape inside the object(shape)

Reply:
Okay. Got it. Thanks